File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
         // storing the freq of numbers
         int count[102] = {0};
        
        for(int num:nums){
            count[num]++;
        }
        
        int totalCount = 0;
        
        for(int i:count){
            totalCount += ((i) * (i-1)/2);
            // ((i) * (i-1)/2) this will give the nc2 = n!/r! * (n-r)!;
        }
        return totalCount;
        
        
    }
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int n = nums.length;
        int goodPair = 0;
        for(int i = 0;i<n-1;i++){
            for(int j = i+1;j<n;j++){
                if(nums[i] == nums[j]){
                    goodPair+=1;
                }
            }
        }
        return goodPair;
    }
}